### DISCUSSION TITLE
使用monorepo管理umi项目群时，使用公共dva model和简易数据流model的最佳实践是什么？

### DISCUSSION THREAD

**Discussion by aladdinLee** [umijs/umi#10886] [Q&A]
众所周知，不论是使用dva还是简易数据流，其model文件都创建在本项目src/models目录下。
对于一些公共逻辑显然抽离并封装到单独的库里更合适一些。
目前想到的方法仅仅是先把model文件抽离出来，然后再在使用到的项目中创建软链。
不知有没有更合适的方法？

**Comment by fz6m** [comment]
`dva` 不维护了，不推荐使用。现在推荐使用 [`valtio`](https://umijs.org/docs/max/valtio) 或 [jotai](https://github.com/pmndrs/jotai) 管理全局数据。

**Reply by semmywong** [reply]
> `dva` 不维护了，不推荐使用。现在推荐使用 [`valtio`](https://umijs.org/docs/max/valtio) 或 [jotai](https://github.com/pmndrs/jotai) 管理全局数据。

我也尝试使用了[valtio](https://umijs.org/docs/max/valtio)，也不能解决上面的问题

**Comment by semmywong** [comment]
> 众所周知，不论是使用dva还是简易数据流，其model文件都创建在本项目src/models目录下。 对于一些公共逻辑显然抽离并封装到单独的库里更合适一些。 目前想到的方法仅仅是先把model文件抽离出来，然后再在使用到的项目中创建软链。 不知有没有更合适的方法？

我也想了解下，你目前创建软链接是怎么创建的？

**Comment by semmywong** [comment]
> 众所周知，不论是使用dva还是简易数据流，其model文件都创建在本项目src/models目录下。 对于一些公共逻辑显然抽离并封装到单独的库里更合适一些。 目前想到的方法仅仅是先把model文件抽离出来，然后再在使用到的项目中创建软链。 不知有没有更合适的方法？

我尝试在使用的项目中重新export default抽取出来的models发现也不能生效
